File: C#/1480_RunningSumArray1D.cs
using System;

namespace RunningSum1D
{
    public class Solution
    {
        public static int[] RunningSum(int[] nums)
        {
            int i = 0;
            int j = 0;
            int[] result = { 0, 0, 0, 0 };
            foreach (int nb in nums)
            {
                result[i] = nb;
                j = i;
                while (j >= 0)
                {
                    if (j != i)
                        result[i] += nums[j];
                    j--;
                }
                i++;
            }
            return (result);
        }
    }
}

File: C/1480_RunningSumArray1D.c
#include <stdlib.h>
#include <stdio.h>

int *runningSum(int *nums, int numsSize, int *returnSize)
{
	int *result;
	int x;

	x = 0;
	result = (int*)malloc(numsSize * sizeof(int));
	result[x] = nums[x];
	for (x = 1; x < numsSize; x++)
	{
		result[x] = result[x - 1] + nums[x];
	}
	*returnSize = numsSize;
	return (result);
}
